addons/mail/models/models.py

```diff
@@ -98,7 +98,7 @@ def _notify_get_reply_to(self, default=None, records=None, company=None, doc_nam
         :param records: DEPRECATED, self should be a valid record set or an
           empty recordset if a generic reply-to is required;
         :param company: used to compute company name part of the from name; provide
-          it if already known, otherwise fall back on user company;
+          it if already known, otherwise use record's company and fall back on user company;
```

Comment:
-> we use records company it they all belong to the same company, to ease cross-company reply-to management in simple use cases. Would also say that in mixed environments (several different companies) we fallback on user company.